api/src/pcapi/core/finance/api.py

```diff
@@ -1333,6 +1333,41 @@ def cancel_pricing(
     return pricing
 
 
+def update_finance_event_pricing_date(
+    stock: offers_models.Stock | educational_models.CollectiveStock,
+    new_pricing_ordering_date: datetime.datetime,
+) -> None:
+    pricing_point_id = stock.offer.venue.current_pricing_point_id
+    finance_events_from_pricing_point = models.FinanceEvent.query.options(
+        sqla.orm.joinedload(models.FinanceEvent.pricings)
+    ).filter(models.FinanceEvent.pricingPointId == pricing_point_id)
+
+    bookings_of_this_stock = bookings_models.Booking.query.with_entities(bookings_models.Booking.id).filter(
+        bookings_models.Booking.stockId == stock.id
+    )
+    finance_events_from_stock = finance_events_from_pricing_point.filter(
+        models.FinanceEvent.bookingId.in_(bookings_of_this_stock)
+    ).all()
+
+    if finance_events_from_stock:
+        old_pricing_ordering_date = finance_events_from_stock[0].pricingOrderingDate
+        for finance_event in finance_events_from_stock:
+            finance_event.pricingOrderingDate = new_pricing_ordering_date
+            db.session.add(finance_event)
+        first_event = (
+            models.FinanceEvent.query.filter(
+                models.FinanceEvent.pricingPointId == pricing_point_id,
+                models.FinanceEvent.pricingOrderingDate >= min(old_pricing_ordering_date, new_pricing_ordering_date),
```

Comment:
... suite aux commentaires plus haut, cette ligne pourra alors être modifiée en : 

```suggestion
                models.FinanceEvent.pricingOrderingDate >= oldest_pricing_ordering_date,
```